src/ui/common/src/components/workflows/workflowHeader.tsx

```diff
@@ -89,17 +91,45 @@ const WorkflowHeader: React.FC<Props> = ({ user, workflowDag }) => {
     [key: string]: string;
   }>({});
 
+  // Returns the map of parameters, from name to serialized spec (which includes the
+  // base64-encoded value and serialization_type).
+  const serializeParameters = () => {
+    let serializedParams = {}
+    Object.entries(paramNameToValMap).forEach(([key, strVal]) => {
+      // Serialize the user's input string appropriately into base64. The input can either be a
+      // 1) number 2) string 3) json.
+      try {
+        let val = JSON.parse(strVal)
+
+        // All jsonable values are serialized as json.
+        serializedParams[key] = JSON.stringify({
+          val: btoa(JSON.stringify(val)),
```

Comment:
could you just use strval here instead of having to stringify val back again?

```typescript
val: btoa(strVal)
```

Reply:
yep, good catch. Fixed.